http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -339,33 +343,47 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
     ev: Ask[F, AppContext]
   ): F[(ContainerName, WsmControlledResourceId)] = {
     val stagingContainerName = ContainerName(s"ls-${params.runtime.runtimeName.asString}")
-    val storageContainerCommonFields = getCommonFields(
-      ControlledResourceName(s"c-${stagingContainerName.value}"),
-      "leonardo staging bucket",
-      params.runtime.auditInfo.creator,
-      None
-    )
     for {
       ctx <- ev.ask[AppContext]
-      resp <- wsmDao.createStorageContainer(
-        CreateStorageContainerRequest(
-          params.workspaceId,
-          storageContainerCommonFields,
-          StorageContainerRequest(stagingContainerName)
-        ),
-        auth
+      wsmApi <- buildWsmControlledResourceApiClient
+      common = getCommonFieldsForWsmGeneratedClient(ControlledResourceName(stagingContainerName.value),
+                                                    "leonardo staging bucket",
+                                                    params.runtime.auditInfo.creator
       )
+      azureStorageContainer = new AzureStorageContainerCreationParameters()
+        .storageContainerName(stagingContainerName.value)
+
+      request = new CreateControlledAzureStorageContainerRequestBody()
+        .common(common)
+        .azureStorageContainer(azureStorageContainer)
+      storageContainer <- F
+        .delay(wsmApi.createAzureStorageContainer(request, params.workspaceId.value))
+        .handleErrorWith(err =>
```

Comment:
I don't think this is correct for two reasons:
-  `handleErrorWith` in the `LeoPubsubMessageSubscriber` around `createAndPollRuntime` call will already handle any errors synchronous to the function call with the handler `handleAzureRuntimeCreationError`. If you really want to, you can `adaptError` here and make the error message `"Wsm createStorageContainer failed due to ${err.getMessage}"`, but that's not strictly necessary since the error message from WSM _should_ have information on what call failed. In other words, neither the `handleErrorWith`  or the `F.raiseError` are necessary here.
- Shouldn't we be polling for the storage container result in the async task formed in `monitorCreateRuntime`, as this TODO states: https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala#L174

Reply:
yeah I see where you're coming from here, this was an attempt to keep the storage container creation out of the async task because the output of `createStorageContainer` is necessary for the createVmRequest --> ie then I'd have to move the createVmRequest into the async as well, but I went ahead and moved all that + creating the hybrid connection into the async task so all the sub-resources could be created there